Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -832,6 +832,118 @@ def list_device_groups_names():
+    """
+    return_results('Starting TSF export...')
+    params = {
+        'type': 'export',
+        'category': 'tech-support',
+        'key': API_KEY
+    }
+    result = http_request(
+        URL,
+        'GET',
+        params=params
+    )
+    return result
+
+
+def get_tsf_export_status(job_id: str):
+    """
+    Get status of TSF export.
+    """
+    return_results('Getting TSF export status...')
+    params = {
+        'type': 'export',
+        'category': 'tech-support',
+        'action': 'status',
+        'job-id': job_id,
+        'key': API_KEY
+    }
+    result = http_request(
+        URL,
+        'GET',
+        params=params
+    )
+    return result
+
+
+def download_tsf(job_id: str):
+    """
+    Download an exported TSF.
+    """
+    params = {
+        'type': 'export',
+        'category': 'tech-support',
+        'action': 'get',
+        'job-id': job_id,
+        'key': API_KEY
+    }
+    result = http_request(
+        URL,
+        'GET',
+        params=params,
+        is_file=True
+    )
+    return fileResult("tech_support_file.tar.gz", result.content)
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get('interval_in_seconds', 60)),
+    timeout=arg_to_number(demisto.args().get('timeout', 1200))
+)
+def export_tsf_command(args: dict):
+    """
+    Export a TSF from PAN-OS.
+    """
+    if job_id := args.get('job_id'):
+        export_tsf_status = get_tsf_export_status(job_id).get('response', {}).get('result', {})
+        job_status = export_tsf_status.get('job', {}).get('status')
+        context_output = {
+            'JobID': job_id,
+            'Status': job_status
+        }
+        return PollResult(
+            response=download_tsf(job_id),
+            continue_to_poll=job_status != 'FIN',  # continue polling if job isn't done
+        )
+    else:  # either no polling is required or this is the first run
+        result = start_tsf_export()
+        job_id = result.get('response', {}).get('result', {}).get('job', '')
+        if job_id:
+            context_output = {
+                'JobID': job_id,
+                'Status': 'Pending'
+            }
+            continue_to_poll = True
+            export_tsf_output = CommandResults(  # type: ignore[assignment]
+                outputs_prefix='Panorama.TSF',
+                outputs_key_field='JobID',
+                outputs=context_output,
+                readable_output=tableToMarkdown('Export Status:', context_output, removeNull=True)
+            )
+        else:  # no job ID which is unexpected
+            raise DemistoException("Failed to start tech support file export.")
+
+        return PollResult(
+            response=export_tsf_output,
+            continue_to_poll=continue_to_poll,
+            args_for_next_run={
+                'job_id': job_id,
+                'polling': argToBoolean(args.get('polling')),
```

Comment:
What is the purpose of this `polling` parameter? I don't see it anywhere in use inside the `export_tsf_output` besides here (which is used for passing a parameter for this function).

Reply:
I guess I thought it was either the convention or a requirement for a polling function. But in this case I think it makes sense to remove the argument and use polling no matter what for this command. Let me know what you think.

Reply:
Removing this arg so that polling happens no matter what.